kmod/src/net.c

```diff
@@ -948,22 +948,27 @@ static int sock_opts_and_names(struct scoutfs_net_connection *conn,
 		goto out;
 
 	addrlen = sizeof(struct sockaddr_in);
-	ret = kernel_getsockname(sock, (struct sockaddr *)&conn->sockname,
+	ret = kc_kernel_getsockname(sock, (struct sockaddr *)&conn->sockname,
 				 &addrlen);
 	if (ret == 0 && addrlen != sizeof(struct sockaddr_in))
 		ret = -EAFNOSUPPORT;
-	if (ret)
+	if (ret < 0)
 		goto out;
 
 	addrlen = sizeof(struct sockaddr_in);
-	ret = kernel_getpeername(sock, (struct sockaddr *)&conn->peername,
+	ret = kc_kernel_getpeername(sock, (struct sockaddr *)&conn->peername,
 				 &addrlen);
 	if (ret == 0 && addrlen != sizeof(struct sockaddr_in))
 		ret = -EAFNOSUPPORT;
-	if (ret)
+	if (ret < 0)
 		goto out;
 
 	conn->last_peername = conn->peername;
+	/*
+	 * Because getsock/peername return (>0) as success, we cannot just
+	 * return ret here.
+	 */
+	return 0;
```

Comment:
The point of the out: nonsense is to have one return statement (that we can then hook into or trace or whatever).  The pattern here is to set ret to 0 before it falls into the label and return, it's all over the place.  Just drop the comment.